### PR TITLE
Support write orc file by custom time zone

### PR DESCRIPTION
**Pull request by KevinyhZou**
ORC format file write always use the default `GMT` time zone, when we write a `dateTime64` type field in a local time zone (like `Asia/Shanghai`), it will make the write value wrong.

In my case, when write a value with local timezone `2024-10-09 20:00:00`, but it written the value as `2024-10-09 12:00:00`, related to issue: https://github.com/apache/incubator-gluten/issues/7502

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support write orc file by custom time zone, not always by `GMT` time zone

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Core/SettingsChangesHistory.cpp [hunk 1]
```diff
@@ -99,6 +99,7 @@ static std::initializer_list<std::pair<ClickHouseVersion, SettingsChangesHistory
             {"distributed_cache_data_packet_ack_window", 5, 5, "A setting for ClickHouse Cloud"},
             {"allow_experimental_refreshable_materialized_view", false, true, "Not experimental anymore"},
             {"max_parts_to_move", 1000, 1000, "New setting"},
+            {"input_format_orc_writer_time_zone_name", "GMT", "GMT", "The time zone name for ORC writer, the default ORC writer's time zone is GMT."},
```
Comment (alexey-milovidov) [inline src/Core/SettingsChangesHistory.cpp:102]: It's output, not input.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit b93567a7abd37723fe7ba4b167ed8b99083fcbcf with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70615/b93567a7abd37723fe7ba4b167ed8b99083fcbcf/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/df4e80bd01c4baaba95cd9471a7f6b95ac5db465/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Core/SettingsChangesHistory.cpp (hunk 1 above)
Comment (KevinyhZou) [reply src/Core/SettingsChangesHistory.cpp]: fixed

**Review by alexey-milovidov — APPROVED** [review]

**Comment by KevinyhZou** [comment]
could you help to merge this pr @alexey-milovidov

**Comment by alexey-milovidov** [comment]
Yes, this is approved, but a few unrelated tests have failed. We need to resolve it...

**Comment by alexey-milovidov** [comment]
@KevinyhZou, are you sure it is GMT, not UTC?
Let's check in the ORC docs.

**Comment by KevinyhZou** [comment]
> @KevinyhZou, are you sure it is GMT, not UTC? Let's check in the ORC docs.

Yes, orc writer time zone default value is `GMT`, and this can be found at the code of orc `Writer.cpp`: https://github.com/apache/orc/blob/d812403d5590c377d486589a9172bb56e5dff54d/c%2B%2B/src/Writer.cc#L68 @alexey-milovidov

**Comment by clickgapai** [comment]
Hi @KevinyhZou @alexey-milovidov — while reviewing this PR I found the following:

- **Bug (P1):** `output_format_orc_writer_time_zone_name` causes DateTime64 timestamp corruption due to double UTC conversion — https://github.com/ClickHouse/ClickHouse/issues/104641

Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**